### ISSUE TITLE
Closing Catalog Modal does not return the focus to the opening button

### ISSUE THREAD

**Issue by pinussilvestrus** [camunda/camunda-modeler#2942]
__Describe the Bug__

When closing the "Catalog" modal, the focus is not returned to the button that opens the modal ("Select" inside the properties panel).

![Kapture 2022-05-06 at 13 32 02](https://user-images.githubusercontent.com/9433996/167123685-f92907b4-b1e3-4a13-a6d2-154facba7ac3.gif)

Without restoring the focus to the opening button, the logical navigation order gets lost. After I close the button, I want to continue navigating from the point where I left.

__Steps to Reproduce__

1. Navigate via keyboard to the "+ Select" button
2. Press enter -> opens Catalog modal
3. Close Catalog modal
4. Focus is lost (set somewhere else) -> I have to navigate back manually from the beginning

__Expected Behavior__

After closing the modal, the focus should be moved back to the "invoking context" previously stored (the Select button).

__Environment__

- OS: MacOS 11.6.5
- Camunda Modeler Version: 5.0.0

**Comment by nikku** [comment]
Idea: Have a "modal" component that we can build upon to handle this.
